controllers/basic_controller.go

```diff
@@ -337,6 +338,9 @@ func (i instanceReconcilerHelper) finalize(ctx context.Context, o client.Object)
 		} else if aiven.IsNotFound(err) {
 			i.rec.Event(o, corev1.EventTypeWarning, eventUnableToDeleteAtAiven, err.Error())
 			return ctrl.Result{}, fmt.Errorf("unable to delete instance at aiven: %w", err)
+		} else {
```

Comment:
It may make more sense to refactor this using a switch case, WDYT.

Reply:
Makes sense, but git blame then will show my changes instead of original PR with feature/fix. So I better keep it that way :)